### PR TITLE
[google_sign_in_platform_interface] Add missing docs

### PR DESCRIPTION
**Pull request by mklim**
## Related Issues

flutter/flutter#45440

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by mklim** [comment]
@ditman I inferred the meanings of a lot of these interfaces by looking at the official GoogleSignInAccount docs for Android and making educated decisions about what paralleled what. Can you verify what I've written here?

I also couldn't figure out what `GoogleSignInTokenData` was supposed to be. Do you mind explaining that also?

File: packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart [hunk 1]
```diff
@@ -4,15 +4,37 @@
 
 import 'package:quiver/core.dart';
 
-enum SignInOption { standard, games }
+/// Holds default configuration options to use when signing in.
+enum SignInOption {
```
Comment (ened) [inline packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart]: Should these options link to https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInOptions.html#public-static-final-googlesigninoptions-default_games_sign_in
and 
https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInOptions.html#DEFAULT_GAMES_SIGN_IN

Obviously, developers that use this plugin should already know what they need, so it really is just whether we could link to it. I'm not familiar with Google Sign In and would then wonder what happened to the user ID once logged in with the Games Sign In.

Comment (mklim) [reply packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart]: Done.

**Comment by mklim** [comment]
In the latest commit I also realized that some of the missing members were documented in the Android/iOS implementation already, so I copied over the pre-existing docs from there to the interface plugin too.

**Comment by ditman** [comment]
Thanks for improving these docs @mklim!!

**Pull request merged**